plugins/search-backend/src/service/router.ts

```diff
@@ -73,6 +71,7 @@ const allowedLocationProtocols = ['http:', 'https:'];
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
+  const router = await initializeRouter();
```

Comment:
I just found a bit confused having an auto spec validation and a manual validation on lines [144 - 148](https://github.com/backstage/backstage/blob/14bf9dbe97171b9c06ab617cd153aa43c1bd3533/plugins/search-backend/src/service/router.ts#L144-L148). Would it be a possibility for the initializer to receive refined validations and execute them when the endpoint is requested? Something like:

```ts
const router = await initializeRouter({
  '/query': {
    requestSchema: queryRequestSchema
  }
});
```

Reply:
Currently, it's not possible with [the library I'm using](https://github.com/cdimascio/express-openapi-validator) but it's definitely an interesting use case. The schema you referenced is one that requires extra run time information about valid `types` and `maxPageSize`, so run time validation would be interesting.

It would definitely give a more consistent user experience for input errors and handling more complex run time schemas. I don't think it's a blocker for this work, but definitely worth exploring in future.

Reply:
Awesome, hope to see contributing to us again when you get a change 🙂